### PR TITLE
[tampermonkey] Tampermonkey 4.8

### PR DESCRIPTION
**Pull request by nikolay-borzov**
If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/tampermonkey/tslint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{ 
+  "extends": "dtslint/dt.json",
+  "rules": {
+    "no-unnecessary-generics": false
```
Comment (weswigham) [inline types/tampermonkey/tslint.json]: I don't see anything that looks like an unneeded generic - why's this lint rule need to be ignored?

Comment (nikolay-borzov) [reply types/tampermonkey/tslint.json]: It's here
```
declare function GM_xmlhttpRequest<TContext = any>(
    details: Tampermonkey.Request<TContext>
): Tampermonkey.AbortHandle<void>;
```

Use case
```
interface RequestContext {
    form: {
        name: string
    };
}

GM_xmlhttpRequest<RequestContext>({
    url: 'http://example.com/',
    onloadstart(response) {
        GM_log(response.context.form.name);
    },
});
```

Comment (weswigham) [reply types/tampermonkey/tslint.json]: I'd prefer to see an inline suppression on that instance than a blanket suppression in the `.json` file, if you could, then - that way future editors don't accidentally add more they don't need to.

Comment (nikolay-borzov) [reply types/tampermonkey/tslint.json]: Sure, I'll add an inline suppression

**Pull request merged**